[DOCS] Added descriptions to Line2

Xgor opened the pull request:
This is my first commit to an Open Source project so I'm sure I got something wrong. Anyways I added information to all methods, members and constants. Hopefully, the descriptions are up to snuff.

Xgor committed:
Added documenation for Line2D

Xgor committed:
Updated with some better description

Xgor committed:
Fixed most of the suggestions

I'll do a more through checkup tomorrow and answering comments but I think i fixed most of the problems.

NathanLovato commented on doc/classes/Line2D.xml:
```diff
@@ -17,7 +17,7 @@
 			</argument>
 			<description>
-				Add a point at the x/y position in the supplied [Vector2]
+				Add a point at the x/y position in the supplied [Vector2]. The point will be added at the end of the line.
```
Don't use passive voice. See the [writing guidelines](http://docs.godotengine.org/en/latest/community/contributing/docs_writing_guidelines.html). The original description (the first sentence) also needs a fix. Example: `Add a point at the [code]position[/code]. Appends the point at the end of the line.`
_Note: I'm not sure of the description. Is there a limit to the number of points you can add?_

Noshyaar replied:
Just noticed this one, I thought x/y meant 'x divided by y' by skimming through. Might be better to use (x, y) or just 'x and y'.

NathanLovato commented on doc/classes/Line2D.xml:
```diff
@@ -210,44 +213,63 @@
+			The style used for the shape in the first point of the line.
 		</member>
 		<member name="default_color" type="Color" setter="set_default_color" getter="get_default_color">
+      The color of the Line2D. Will not be used if a gradient is set.
```
The line needs 2 extra indents

Noshyaar replied:
Or use tabs 😛

NathanLovato commented on doc/classes/Line2D.xml:
```diff
@@ -210,44 +213,63 @@
 	</methods>
 	<members>
 		<member name="begin_cap_mode" type="int" setter="set_begin_cap_mode" getter="get_begin_cap_mode" enum="Line2D.LineCapMode">
+			The style used for the shape in the first point of the line.
```
Don't use passive voice (used), try to use the direct tense instead, and use `'s` to mark possession:
`Controls the style of the line's first point.`

NathanLovato commented on doc/classes/Line2D.xml:
```diff
@@ -54,6 +54,7 @@
 			<return type="int">
 			</return>
 			<description>
+				Return the amount of points in the Line2D.
```
The method does the action:
`Return` -> `Returns`
Use [code] markup for Classes etc. you don't want to link to, and [ClassName] to generate a link.
`[code]Line2D[/code]

NathanLovato reviewed (changes requested):
Don't worry about getting something wrong: we all do! It's a pleasure to see another contributor. Drop by on Discord for live help (see below)
When you modify your initial commit, please use `git commit --amend` (just in case, type `:wq` to save and quit in vim, the text editor that'll open in the code). Now you need to squash or fixup your commits (merge them into 1) before someone will accept them.

Please review the [writing guidelines](http://docs.godotengine.org/en/latest/community/contributing/docs_writing_guidelines.html) and the [contribution guide](http://docs.godotengine.org/en/latest/community/contributing/updating_the_class_reference.html)

Drop by the #documentation channel [on Discord](https://discord.gg/8FE9RmX), where we run weekly docs sprints.

cbscribe commented on doc/classes/Line2D.xml:
```diff
@@ -62,6 +63,7 @@
 			<argument index="0" name="i" type="int">
 			</argument>
 			<description>
+				Return the x/y position of point 'i'.
```
Return -> Returns

Positions innately have x and y coordinates in 2D, so I don't think it's necessary to mention them. You also need to say what "i" is referring to:

    Returns point [code]i[code]'s position.

Use [code][/code] tags for highlighting variables, etc.

cbscribe commented on doc/classes/Line2D.xml:
```diff
@@ -210,44 +213,63 @@
+      The color of the Line2D. Will not be used if a gradient is set.
 		</member>
 		<member name="end_cap_mode" type="int" setter="set_end_cap_mode" getter="get_end_cap_mode" enum="Line2D.LineCapMode">
+			The style used for the shape in the last point of the line.
```
Too many prepositions. Instead of "foo of the bar" just use "bar's foo".
Typically, it's helpful to note the default value. 
Try something like:

    The final point's style. Use [code]LINE_CAP_*[/code] constants. Default value: [code]LINE_CAP_NONE[/code].

cbscribe commented on doc/classes/Line2D.xml:
```diff
@@ -210,44 +213,63 @@
+			The points that form the lines. The line is drawn between every point set in this array.
 		</member>
 		<member name="round_precision" type="int" setter="set_round_precision" getter="get_round_precision">
+			Sets how smooth the rounded joints and caps will be. This is only used if a cap or joint is set as round.
```
Don't use sets/gets for member variables - they do both.  "Smoothness" or "Smoothing" might work.

cbscribe commented on doc/classes/Line2D.xml:
```diff
@@ -210,44 +213,63 @@
+			This sets the direction difference in radians between vector points. This value is only used if [code]joint mode[/code] is set to [code]LINE_JOINT_SHARP[/code]
 		</member>
 		<member name="texture" type="Texture" setter="set_texture" getter="get_texture">
+			The texture used for the line. The way it's rendered on the line is set in [code]texture_mode[/code].
```
The line's texture. Uses [code]texture_mode[/code] for drawing style.

cbscribe commented on doc/classes/Line2D.xml:
```diff
@@ -210,44 +213,63 @@
+			The joints of the line will be pointy. If Sharp Limit is greater than the rotation of a joint it becomes a bevel joint instead.
 		</constant>
 		<constant name="LINE_JOINT_BEVEL" value="1">
+			The joints of the line
```
Incomplete?

cbscribe commented on doc/classes/Line2D.xml:
```diff
@@ -210,44 +213,63 @@
+			The joints of the line
 		</constant>
 		<constant name="LINE_JOINT_ROUND" value="2">
+			The joints of the line is rounded
```
Incomplete sentence and is -> are

cbscribe commented on doc/classes/Line2D.xml:
```diff
@@ -210,44 +213,63 @@
+			The joints of the line is rounded
 		</constant>
 		<constant name="LINE_CAP_NONE" value="0">
+			Don't have a line cap
```
Missing period.  Follow your same pattern: "Draws the line with no end cap."

cbscribe commented on doc/classes/Line2D.xml:
```diff
@@ -210,44 +213,63 @@
+			Draws the line cap as a circle.
 		</constant>
 		<constant name="LINE_TEXTURE_NONE" value="0">
+			Takes the left pixels of the texture and renders it over the whole line.
```
This description is confusing.

Xgor committed:
Fixed most of the suggestions

I think I got most of it but I'll look more trough tomorrow and I'll answer the comments.

NathanLovato commented:
There are some conflict messages (>>> HEAD) in the PR, and you need to squash your commits - we can't merge 4 commits that should be one. And you must rebase your changes, see the contribution guide I left in the last review comment.

StraToN commented on doc/classes/Line2D.xml:
```diff
@@ -210,44 +217,76 @@
+			The points that form the lines. The line is drawn between every point set in this array.
 		</member>
 		<member name="round_precision" type="int" setter="set_round_precision" getter="get_round_precision">
+<<<<<<< HEAD
```
You seem to have an unresolved conflict here.

StraToN commented on doc/classes/Line2D.xml:
```diff
@@ -210,44 +217,76 @@
 	</members>
 	<constants>
 		<constant name="LINE_JOINT_SHARP" value="0">
+			The joints of the line will be pointy. If Sharp Limit is greater than the rotation of a joint it becomes a bevel joint instead.
```
According to [documentation writing guidelines](http://docs.godotengine.org/en/latest/community/contributing/docs_writing_guidelines.html#use-s-to-show-possession), one should write "line's joints" instead of "joints of the line". 
To apply everywhere.

akien-mga commented:
Merged manually as 17011f5d12 with some of the suggestions above.
